### ISSUE TITLE
PHP Fatal error:  Cannot declare class error, because the name is already in use in 

### ISSUE THREAD

**Issue by jcubic** [rvanvelzen/lime#2]
I got this error in line 161 when I try to call:

```
./lime.php examples/calc/calc.lime > parser.php
```

I'm using php version 7.0.8

**Comment by rvanvelzen** [comment]
Yes, in PHP 7 this class (`Error`) is already defined. I'll modernize the code soon

**Comment by AlgorithMan-de** [comment]
I got the same problem after "upgrading" to Linux Mint 18 :-(

**Comment by AlgorithMan-de** [comment]
I think this patch might solve the issue - I have just moved everything in a namespace (php namespaces are a bit stupid though). I can't test it right now - I'm currently on a Linux Mint 17.3 machine that has php 5.5. It should be noted, that namespaces require php >= 5.3

[limeNamespacePatch.txt](https://github.com/rvanvelzen/lime/files/610908/limeNamespacePatch.txt)

**Comment by AlgorithMan-de** [comment]
I think this patch might solve the issue - I have just moved everything in a namespace (php namespaces are a bit stupid though). I can't test it right now - I'm currently on a Linux Mint 17.3 machine that has php 5.5. It should be noted, that namespaces require php >= 5.3

To apply the patch, just run `git apply limeNamespacePatch.txt` inside the lime directory.
[limeNamespacePatch.txt](https://github.com/rvanvelzen/lime/files/610937/limeNamespacePatch.txt)

**Comment by gene-sis** [comment]
I added namespaces with PSR-4 on my Lime fork and made it work with composer (and fixed some windows related issues). I'm not that experienced, but maybe you want to take a look at it anyway.

@AlgorithMan-de I think my first approach was similar to your patch and I had the problem, that the metaparser didn't work anymore. (I haven't tested your patch, so I don't know if it would have the same problem)

**Comment by AlgorithMan-de** [comment]
I just tested it and it works :-)

**Comment by AlgorithMan-de** [comment]
@rvanvelzen do you want to merge it? I can also fork it, merge it myself and request a pull if you prefer that.
